ports/glutin/window.rs

```diff
@@ -364,40 +410,62 @@ impl Window {
-                                     .get()
-                                     .and_then(|ch| filter_nonprintable(ch, virtual_key_code));
-                        self.pending_key_event_char.set(None);
-                        if let Some(ch) = ch {
-                            self.pressed_key_map.borrow_mut().push((scan_code, ch));
+                #[cfg(not(target_os = "windows"))]
+                {
+                    let ch = match element_state {
+                        ElementState::Pressed => {
+                            // Retrieve any previously stored ReceivedCharacter value.
+                            // Store the association between the scan code and the actual
+                            // character value, if there is one.
+                            let ch = self.pending_key_event_char
+                                        .get()
+                                        .and_then(|ch| filter_nonprintable(ch, virtual_key_code));
+                            self.pending_key_event_char.set(None);
+                            if let Some(ch) = ch {
+                                self.pressed_key_map.borrow_mut().push((_scan_code, ch));
+                            }
+                            ch
                         }
-                        ch
-                    }
 
-                    ElementState::Released => {
-                        // Retrieve the associated character value for this release key,
-                        // if one was previously stored.
-                        let idx = self.pressed_key_map
-                                      .borrow()
-                                      .iter()
-                                      .position(|&(code, _)| code == scan_code);
-                        idx.map(|idx| self.pressed_key_map.borrow_mut().swap_remove(idx).1)
+                        ElementState::Released => {
+                            // Retrieve the associated character value for this release key,
+                            // if one was previously stored.
+                            let idx = self.pressed_key_map
+                                        .borrow()
+                                        .iter()
+                                        .position(|&(code, _)| code == _scan_code);
+                            idx.map(|idx| self.pressed_key_map.borrow_mut().swap_remove(idx).1)
+                        }
+                    };
+
+                    if let Ok(key) = Window::glutin_key_to_script_key(virtual_key_code) {
+                        let state = match element_state {
+                            ElementState::Pressed => KeyState::Pressed,
+                            ElementState::Released => KeyState::Released,
+                        };
+                        let modifiers = Window::glutin_mods_to_script_mods(self.key_modifiers.get());
+                        self.event_queue.borrow_mut().push(WindowEvent::KeyEvent(None, key, state, modifiers));
```

Comment:
Yes, it is to align the behavior between platform. In Windows, we get the KeyboardInput event before the ReceivedCharacter which lead to a KeyEvent being fired with no character until we get the associated ReceivedCharacter (if any). In the other OSs the order is reversed. With that line, we simulate reorder the events on non Windows platforms.  Said differently, you can see there is a KeyEvent fired in both ReceivedCharacter and KeyboardInput in Windows, while no event are fired in the ReceivedCharacter branch for other platform, we force us to fire 2 event in the KeyboardInput branch to get the same behavior.